tests/fake/test_fake_remove_content.py

```diff
@@ -100,3 +106,77 @@ def test_remove_deleted_repo_fails():
 
     # It should fail
     assert "Repository id=repo1 not found" in str(remove.exception())
+
+
+def test_remove_filtered_content():
+    """repo.remove() succeeds and removes expected units inserted via controller."""
+    controller = FakeController()
+    client = controller.client
+
+    rpm_units = [
+        RpmUnit(name="bash", version="4.0", release="1", arch="x86_64"),
+        RpmUnit(name="glibc", version="5.0", release="1", arch="x86_64"),
+    ]
+    modulemd_units = [
+        ModulemdUnit(
+            name="module1", stream="s1", version=1234, context="a1b2", arch="x86_64"
+        ),
+        ModulemdUnit(
+            name="module1", stream="s1", version=1235, context="a1b2", arch="x86_64"
+        ),
+    ]
+    units = rpm_units + modulemd_units
+
+    repo = YumRepository(id="repo1")
+    controller.insert_repository(repo)
+    controller.insert_units(repo, units)
+
+    filters = Criteria.with_field("name", "bash")
+    remove_rpms = client.get_repository("repo1").remove_content(
```

Comment:
Could we please have at least one case which calls remove_content with criteria and without type_ids ?  (which I'm saying elsewhere should now be the preferred usage)